File: Programming_Languages/asgn2/python_code/task3_java/SmartFixture.java
package assg2task3;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class SmartFixture extends Fixture {
    protected double battery_capacity;
    protected double battery_remaining;
    public SmartFixture(String Name, double Weight, int Position, double Battery_capacity){
        super(Name, Weight, Position);
        this.battery_capacity = Battery_capacity;
        this.battery_remaining = Battery_capacity;
    }
    public void ChargeBattery(double Amount){
        this.battery_remaining += Amount;
        if(this.battery_remaining >this.battery_capacity){
            this.battery_remaining = this.battery_capacity;
        }
        System.out.println(this.name + " was charged " + Amount + " units of battery." );
    }
    public double ConsumeBattery(double Amount){
        if(Amount >= this.battery_remaining){
            return -1;
        }
        else{
            this.battery_remaining -= Amount;
            return this.battery_remaining;
        }
    }
    public void MoveTo(Room Destination){
        int distance = Math.abs(this.position - Destination.GetPosition());
        double battery_consumed = distance * this.weight;
        if (this.ConsumeBattery(battery_consumed) > 0){
            this.position = Destination.GetPosition();
            System.out.println(this.name + " moved to " + Destination.GetName() + ".");
            System.out.println(this.name + " consumed " + battery_consumed + " battery." );
        }
        else{
            System.out.println("Not enough battery to move!");
        }
    }
}

File: Programming_Languages/asgn2/python_code/task3_python/SmartFixture.py
#this is the python smart Fixture

import abc
import Fixture
#import Attendant
#import Room

class SmartFixture(Fixture):

    def __init__(self, Name, Weight, Position, Battery_capacity):
        Fixture.__init__(self,Name, Weight, Position)
        self._battery_capacity = Battery_capacity
        self._battery_remaining = Battery_capacity

    def ChargeBattery(self, Amount):
        self._battery_remaining = self._battery_remaining+Amount
        if (self._battery_remaining>self._battery_capacity):
            self._battery_remaining=self._battery_capacity
        print ('%s was charged %.2f units of battery' % self._name, Amount)

    def ConsumeBattery(self, Amount):
        if(Amount > _battery_remaining):
            return -1
        else:
            self._battery_remaining=self._battery_remaining-Amount
            return sefl._battery_remaining

    def MoveTo(self,Destination):
        distance = abs(self.position - Destination.GetPosition())
        battery_consumed = distance * self._weight
        if(self.ConsumeBattery(battery_consumed)>0):
            self.position=Destination.GetPosition()
            print ('%s moved to %s.' % self._name, Destination.GetName())
            print ('%s consumed %.2f battery.' % self._name, battery_consumed)
        else:
            print ('Not enough battery to move!')
